homeassistant/components/fan/__init__.py

```diff
@@ -29,8 +29,8 @@
 
 ENTITY_ID_FORMAT = DOMAIN + '.{}'
 
-# Bitfield of features supported by the fan entity
-ATTR_SUPPORTED_FEATURES = 'supported_features'
+# Bitfield of functions supported by the fan entity
+ATTR_SUPPORTED_FUNCTIONS = 'supported_functions'
```

Comment:
The problem is that the bitfield doesn't properly work since `SUPPORT_BRIGHTNESS` is `1` as is `SUPPORT_SET_SPEED` which causes the later logic to detect the domain and such not run.

However, having written all that out I realized that I could just do a domain check before checking the bitfield... This is what happens when you program while on extra strength cold medicine. 😜